### PR TITLE
Updated conda CLI help to be better for users.

### PR DESCRIPTION
**Pull request by kathatherine**
### Description

This PR contains the first pass at the checklist in #11662. Most of these changes are rewordings of descriptions or formatting or grammar fixes.

I did mess with argparse a bit, and am having trouble building docs locally, so I'm going to make sure everything looks okay using the docs built by this PR.

### Checklist - did you ...

- [x] ~Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [x] ~Add / update necessary tests?~
- [x] Add / update outdated documentation?

### CODE REVIEW

File: conda/cli/conda_argparse.py [hunk 1]
```diff
@@ -249,7 +249,7 @@ def configure_parser_clean(sub_parsers):
     Remove unused packages and caches.
     """)
     example = dedent("""
-    Examples:
+    Examples::
```
Comment (jezdez) [inline conda/cli/conda_argparse.py:252]: Love this!!

File: conda/cli/conda_argparse.py [hunk 2]
```diff
@@ -843,41 +869,37 @@ def configure_parser_list(sub_parsers):
...
         action="store_true",
-        help="List the revision history and exit.",
+        help="List the revision history.",
     )
-    p.add_argument(
-        "--no-pip",
-        action="store_false",
-        default=True,
-        dest="pip",
-        help="Do not include pip-only installed packages.")
```
Comment (jezdez) [inline conda/cli/conda_argparse.py:880]: Should this be removed?

File: conda/cli/conda_argparse.py [hunk 3]
```diff
@@ -1564,13 +1597,13 @@ def add_parser_channels(p):
...
-        'defaults' to get the default packages for conda.  You can also use any name and the
-        .condarc channel_alias value will be prepended.  The default channel_alias
-        is https://conda.anaconda.org/.""",
+        help=("Additional channel to search for packages. These are URLs searched in the order "
+              "they are given (including local directories using the 'file://'  syntax or "
+              "simply a path like '/home/conda/mychan' or '../mychan').  Then, the defaults "
+              "or channels from .condarc are searched (unless --override-channels is given). You can use "
+              "'defaults' to get the default packages for conda. You can also use any name and the "
+              ".condarc channel_alias value will be prepended. The default channel_alias "
+              "is https://conda.anaconda.org/.")
```
Comment (jezdez) [inline conda/cli/conda_argparse.py:1606]:
```suggestion
        help=("Additional channel to search for packages. These are URLs searched in the order "
              "they are given (including local directories using the 'file://' syntax or "
              "simply a path like '/home/conda/mychan' or '../mychan'). Then, the defaults "
              "or channels from .condarc are searched (unless --override-channels is given). You can use "
              "'defaults' to get the default packages for conda. You can also use any name and the "
              ".condarc channel_alias value will be prepended. The default channel_alias "
              "is https://conda.anaconda.org/.")
```

**Review by jezdez — APPROVED** [review]
Nice!! Just a few nits

File: conda/cli/conda_argparse.py (hunk 2 above)
Comment (kathatherine) [reply conda/cli/conda_argparse.py:880]: Yeah, I wasn't totally sure on that. @kenodegard said it should be deprecated as an argument, so maybe we should just mark it as (deprecated) in here? Or I'll make a ticket to actually properly deprecate the argument and the argparse work can be done then?

Comment (kenodegard) [reply conda/cli/conda_argparse.py:880]: This argument is unused

<details>
<summary>`--no-pip` → `args.pip` → `piplist` and this `piplist` variable is unused</summary>

https://github.com/conda/conda/blob/8b846957c4338188706959756e7dad1619727c59/conda/cli/conda_argparse.py#L875-L880

https://github.com/conda/conda/blob/8b846957c4338188706959756e7dad1619727c59/conda/cli/main_list.py#L139-L141

https://github.com/conda/conda/blob/8b846957c4338188706959756e7dad1619727c59/conda/cli/main_list.py#L68-L86

</details>

But I agree it should go through a deprecation process, not removed outright

Comment (kathatherine) [reply conda/cli/conda_argparse.py:880]: Gotcha gotcha. Will fix!

**Comment by kathatherine** [comment]
pre-commit.ci autofix

File: conda/cli/conda_argparse.py [hunk 4]
```diff
@@ -557,15 +573,21 @@ def configure_parser_config(sub_parsers):
 
 def configure_parser_create(sub_parsers):
     help = "Create a new conda environment from a list of specified packages. "
-    descr = (help + "To use the created environment, use 'conda activate "
-             "envname' to look in that directory first.  This command requires either "
-             "the -n NAME or -p PREFIX option.")
+    descr = (help + "To use the newly-created environment, use 'conda activate "
+             "envname'. This command requires either the -n NAME or - PREFIX"
```
Comment (beeankha) [inline conda/cli/conda_argparse.py:577]:
```suggestion
             "envname'. This command requires either the -n NAME or -p PREFIX"
```

File: conda/cli/conda_argparse.py [hunk 5]
```diff
@@ -576,8 +598,8 @@ def configure_parser_create(sub_parsers):
     p.add_argument(
         "--clone",
         action="store",
-        help='Path to (or name of) existing local environment.',
-        metavar='ENV',
+        help="Create a new environment as a copy of an existing local " "environment.",
```
Comment (beeankha) [inline conda/cli/conda_argparse.py:601]:
```suggestion
        help="Create a new environment as a copy of an existing local environment.",
```

File: conda/cli/conda_argparse.py [hunk 6]
```diff
@@ -951,19 +993,19 @@ def configure_parser_package(sub_parsers):
...
         default="unknown",
-        help="Package name of the created package.",
+        help="Designate package name of the package being created.",
     )
     p.add_argument(
         "--pkg-version",
         action="store",
         default="0.0",
-        help="Package version of the created package.",
+        help="Designated package version of the package being created.",
```
Comment (beeankha) [inline conda/cli/conda_argparse.py:1002]:
```suggestion
        help="Designate package version of the package being created.",
```

**Review by beeankha — APPROVED** [review]
LGTM!

**Pull request merged**